sstables/sstables.cc

```diff
@@ -1280,25 +1280,36 @@ sstable::find_first_position_in_partition(reader_permit permit, const dht::decor
     co_return std::move(ret);
 }
 
-future<> sstable::load_first_and_last_position_in_partition() {
+void sstable::load_first_and_last_position_in_partition() {
     if (!_schema->clustering_key_size()) {
-        co_return;
+        return;
     }
 
-    auto& sem = _manager.sstable_metadata_concurrency_sem();
-    reader_permit permit = co_await sem.obtain_permit(&*_schema, "sstable::load_first_and_last_position_range", sstable_buffer_size, db::no_timeout);
-    auto first_pos_opt = co_await find_first_position_in_partition(permit, get_first_decorated_key(), false);
-    auto last_pos_opt = co_await find_first_position_in_partition(permit, get_last_decorated_key(), true);
-
-    // Allow loading to proceed even if we were unable to load this metadata as the lack of it
-    // will not affect correctness.
-    if (!first_pos_opt || !last_pos_opt) {
-        sstlog.warn("Unable to retrieve metadata for first and last keys of {}. Not a critical error.", get_filename());
-        co_return;
+    auto cpm = _components->scylla_metadata->get_optional_clustering_position_metadata();
+    if (!cpm) {
+        sstlog.debug("Clustering position metadata not found for {}", get_filename());
+        return;
```

Comment:
Should we fall-back to the old method of reading the first/last row? Won't this cause problems with already split sstables that were written before this PR? Or was the splitting code not released yet?

Reply:
partition splitting is not wired anywhere yet. for that to happen, we need to write the run reader first and extend incremental compaction to operate on clustering domain.